### ISSUE TITLE
Revisting a classification approach for distinguishing reads from charged or uncharged molecules

### ISSUE THREAD

**Issue by kriemo** [rnabioco/aa-tRNA-seq-pipeline#16]
The current filtering strategy in the pipeline discards reads that map equivalently to the charged and uncharged tRNA references. This approach confidently identifies charged and uncharged reads with a low false positive rate for most tRNAs (with proline and serine tRNAs as an exception). The stringency of this approach however preferentially results in excluding more charged tRNA alignments than uncharged, resulting in artificially lowered estimates of charging status per tRNA. The charged tRNA molecules have higher error rates at the 3’ end, more reads aligned with MAPQ0, which results in more of these reads being filtered.

An alternative to this post-hoc filtering strategy is a classification based approach, based on the work done by Marcin.  Within the 3’ adapter sequence, there is a 13 base region that differs between charged (`CCAACCTTGCCTT`) and uncharged (`ATGGAAGGTAGGC`). The base calling errors of reads from true charged molecules likely differ than those errors from true uncharged molecules within this region which can be leveraged by a classifier. 

A classification approach could potentially use:  

 - the read sequence information at these 13 positions  
 - read sequence 5’ of this region up to the CCA (which also have charging specific error profiles)  
 - summary statistics about soft-clipped bases (e.g. number of bases and base composition)  
 - information from the alignment (e.g. MAPQ, alignment score)  
 - base qualities  
 - signal level information (dwell and mean current)  

Some of these features will be co-linear and may be necessary to exclude. 

The classifier could be trained in an analogous way to the approach tested by Marcin on the synthetic datasets e.g.:

1) Generate a dataset with a defined mix of reads from ideally true uncharged molecules and reads from true charged molecules. This could be derived from the single-adapter ligation libraries (e.g. only chemo-ligated charged adapter, and only uncharged T4 RNL2 ligation).

2) Align the reads to the tRNA reference that has both charged and uncharged sequences. There is a lot of variation in the positioning of the adapter sequence w.r.t the 3’ end of the read sequence (as evidenced by all of the soft-clipping seen at the 3’ end of reads). Alignment to the full tRNA reference will (hopefully) confidently define the read bases that derive from the adapter.
 
3) Collect various metrics into a matrix. One hot encode the adapter read bases perhaps with an additional insertion and deletion category for each reference position. 

4) Train a XGboost or similar model on a subset of the data and test performance on the remaining reads. Also check the models ability on different isodecoders. If the model performs similarly well across most isodecoders, then one could argue that the tRNA sequence 5’ of CCA isn’t biasing the modeling, so the model might also work well on libraries from other species. 

5) Test on an independent validation set (e.g. the deacylated libraries?). 

6) Identify key features driving classification, which may identify a simpler set of heuristics that can be used instead of a formal model for classification.
